### ISSUE TITLE
Maintainers wanted

### ISSUE THREAD

**Issue by JMPerez** [JMPerez/spotify-web-api-js#114]
## We would like to get contributors who would like to maintain and evolve this library

As the main maintainer of this library, I have been busy lately with other projects and I will not able to keep this library up to date for the foreseeable future. I would love to get support from other contributors that have ideas on how to move forward the library so it's still relevant and useful for developers.

If you are interested in maintaining the library, please add a comment to this issue.

**Comment by adamgrieger** [comment]
I would love to contribute; I'm mainly a TypeScript user and I've used this library quite a bit this past year. I haven't really had experience in creating an API wrapper library, but it's definitely something I'd be interested in (especially since it's Spotify's API)!

**Comment by csantiago132** [comment]
@adamgrieger yes! Sure I’ll help out in whatever way I can. I’m wrapping up my bootcamp now so I’ll be ready to jump in in a  week

**Comment by adamgrieger** [comment]
I added a Discord server link to my fork's README.md where we (and potentially others) can talk about the rewrite.

**Comment by csantiago132** [comment]
@adamgrieger ok, ill update the branch, I went across the code and wow, great job! ill keep digging around and learn the code-base, ill start with seeing the test, i believe thats the easiest way to get to know the codebase

**Comment by lynnpark** [comment]
@adamgrieger @csantiago132 are you guys still actively maintaining? I'd like to try to attack #129

**Comment by csantiago132** [comment]
@lynnpark id like to but havent heard from @JMPerez idk whats the best procedure or approach that he wants

**Comment by adamgrieger** [comment]
I probably won't be maintaining this project anytime in the near future due to work, so it's all yours if you want to take it!

**Comment by lynnpark** [comment]
> @lynnpark id like to but havent heard from @JMPerez idk whats the best procedure or approach that he wants

Is @JMPerez the only owner? Are there anyone else with admin rights?

**Comment by JMPerez** [comment]
@lynnpark I'm the only owner so far

@adamgrieger, @csantiago132 would you be interested in maintaining this project?

**Comment by csantiago132** [comment]
@JMPerez yes id love to!

**Comment by lynnpark** [comment]
@JMPerez I would love to too, if there is room for one more :)

**Comment by DanielFroehlich** [comment]
Hey, I am wondering what is the state of this case and project. Is it still being maintained? 
I am seeing a lot of good PRs also on the original repo (https://github.com/thelinmichael/spotify-web-api-node).

**Comment by adamgrieger** [comment]
If anyone wants to follow my progress, I've been working on an isomorphic wrapper written in TypeScript. Hopefully when it's done, it can serve as an alternative to both `spotify-web-api-js` and `spotify-web-api-node`.

https://github.com/adamgrieger/spotify-web-api-ts

**Comment by BenGu3** [comment]
Just wanted to start a conversation here about a few of these packages mentioned here. I've been hoping to spend some time helping out because I'm using this package in a project. I just want to understand the future of these packages.

As far as `spotify-web-api-js` goes and my understanding, it looks like it may be nice to Typescript-ify this package. That would help with keeping the types and implementation in sync by letting `tsc` create the type definition files. (I'd be glad to help with this).
- I think this may help with #165 
- Would this go against the point of this package as `a lightweight wrapper`? Thoughts @JMPerez?

It seems like [spotify-web-api-node](https://github.com/thelinmichael/spotify-web-api-node) hasn't been touched in a year or two. 

I know as far as Typescript goes, @adamgrieger has made huge progress on [spotify-web-api-ts](https://github.com/adamgrieger/spotify-web-api-ts), which is great because that could be used in place of both `spotify-web-api-js` and `spotify-web-api-node`. I'd be glad to help out there also, though it seems redundant having all of these packages. 
- @JMPerez, if you are okay with Typescript-ifying and `spotify-web-api-ts` was the maintained package of the three since it has so much already done, would you be willing to help maintain that? (I apologize if I'm out of line. I'm hoping to have an open conversation about all of this 😅)

I'd love to hear everyone's thoughts and ideas about all this.

**Comment by JMPerez** [comment]
@BenGu3 Happy to typescript-ify this project, that would help with maintenance. It's painful adding new endpoints due to having to modify many files when one would expect changing the lib + test.

If you can explore this approach, I'd be glad to review the changes and make a decision. The overhead for the library output will be minimal. Once the JS is generated it shouldn't be that large compared to the current bundle, and adding a couple of kBs is good trade-off if that makes it easier to keep updated.

Regarding @adamgrieger's project, the more the merrier. I can't commit to maintaining more projects at the moment, but I can reach out to the Spotify team to try to list it in their list of wrappers for developers to discover.

**Comment by patrick-motard** [comment]
How does this repo relate to https://github.com/thelinmichael/spotify-web-api-node ? When I initially searched for a spotify sdk in nodejs on NPM, that was the version i found.

It looks like this repo is what is actually keeping https://www.npmjs.com/package/spotify-web-api-node up to date. But the links to source code on that page link to https://github.com/thelinmichael/spotify-web-api-node. Maybe those links should be updated to point to this project?

Will https://github.com/adamgrieger/spotify-web-api-ts be merged into this project in the future?

There are several new issues and pull requests being submitted to https://github.com/thelinmichael/spotify-web-api-node but it's not being updated and doesn't get pubished to NPM. IMO it's issues should be closed and/or moved to this project.

**Comment by patrick-motard** [comment]
Ah... I think I see what is going on now. Not sure how I missed this before.
These are two separate projects.
- [spotify-web-api-node](https://www.npmjs.com/package/spotify-web-api-node) created by [thelinmichael](https://github.com/thelinmichael) and maintained by him and [JMPerez](https://github.com/JMPerez) intended to be used in Node. Claims it's also designed to work in the browser. Hasn't been touched in 2 years. Has [an open issue by JMPerez](https://github.com/thelinmichael/spotify-web-api-node/issues/252) looking for maintainers.
- [spotify-web-api-js](https://github.com/JMPerez/spotify-web-api-js) created by JMPerez, for browser use. Has been kept up to date by JMPerez. With potential re-write/improvements coming from https://github.com/adamgrieger/spotify-web-api-ts in the future. ALSO has an open issue by JMPerez looking for maintainers with the same title. 

Given that my project is in Electron, I'm assuming I want to be using the other project. Though it's confusing to me why https://github.com/thelinmichael/spotify-web-api-node claims:
> This is a universal wrapper/client for the Spotify Web API that runs on Node.JS and the browser

If it can be used in the browser, and JMPerez is a maintainer of it why does spotify-web-api-js exist?
